src/parser/esprima_cpp/esprima.cpp

```diff
@@ -3815,19 +3814,26 @@ class Parser {
 
         this->expect(RightParenthesis);
 
+        bool allowFunctionDeclaration = !this->context->strict;
         bool allowLexicalDeclarationBefore = this->context->allowLexicalDeclaration;
         this->context->allowLexicalDeclaration = false;
 
-        ParserBlockContext blockContext;
-        bool shouldOpenBlock = allowFunctionDeclaration && this->matchKeyword(KeywordKind::FunctionKeyword);
-
-        consequent = shouldOpenBlock ? this->parseSingleStatementNodeAsBlockStatement(builder, allowFunctionDeclaration) : this->parseStatement(builder, allowFunctionDeclaration);
+        if (allowFunctionDeclaration && this->matchKeyword(KeywordKind::FunctionKeyword)) {
+            // In non-strict mode, functions can declared as the body of an if statement
```

Comment:
How about write some comment about spec address?
IMO this spec is https://tc39.es/ecma262/#sec-labelled-function-declarations

Reply:
Thanks for the review. Add spec address as you commented.